app/src/scripts/soc/soc.tsx

```diff
@@ -125,22 +129,46 @@ export abstract class SOC_Generic {
             return [];
 
         const upperPhrase: string = phrase.toUpperCase();
+
+        // Check cache first for both broad and specific course searches
+        // Use course code as cache key for specific searches
+        const cacheKey =
+            searchBy === SearchBy.COURSE_CODE
+                ? upperPhrase
+                : `${searchBy}:${upperPhrase}`;
```

Comment:
The `cacheKey` can always be `${searchBy}:${upperPhrase}` as it is a unique identifier.